Source/santad/SNTPolicyProcessor.mm

```diff
@@ -44,6 +45,128 @@ - (instancetype)initWithRuleTable:(SNTRuleTable *)ruleTable {
   return self;
 }
 
+// This method applies the rules to the cached decision object.
+inline SNTCachedDecision *updateCachedDecisionForRule(SNTCachedDecision *cd, 
```

Comment:
I think we should consider a small change in this interface. As implemented, an in-band signal is used (via `SNTEventStateBlockUnknown` or `SNTEventStateAllowUnknown`) to indicate further processing is needed in the caller.

Since we mutate the provided `SNTCachedDecision`, there isn't much need to return it as well. We could isntead return a boolean as to whether or not a decision was able to be obtained:

```
static BOOL updateCachedDecisionForRule(
    SNTCachedDecision *cd,
    SNTRule *rule,
    // SNTClientMode mode, Note: This parameter wouldn't be needed anymore
    bool enableTransitiveRules)
```

I think this will help simplify the logic and be easier to maintain.

If we go this route, it might be good to change the function name. Maybe `decisionForRule`? Or some name that also makes it clear the provided `SNTCachedDecision` is still mutated would be nice.

(Additional comments in this function are made assuming this change is made)